### ISSUE TITLE
[BUG] `npm update -g package@latest` removes all global packages

### ISSUE THREAD

**Issue by mukaschultze** [npm/cli#3175]
### Current Behavior:
Attempting to update a package with a tag using the `update` command globally causes **ALL** global packages to be removed.

### Expected Behavior:
Packages should not be removed when updating global packages.

### Steps To Reproduce:
```sh
npm i -g typescript nodemon @angular/cli # Install some global packages
npm ls -g --depth=0 # This shows the 3 installed packages
npm update -g @angular/cli@latest # Attempt to update one of the packages with a @latest tag
# Oh no! removed 620 packages
npm ls -g --depth=0 # Shows `-- (empty)
```

### Environment:
- OS: Windows 10 20H2
- Node: 15.9.0
- npm: 7.11.2

### Additional Info
This only happens on the global scope

**Comment by wraithgar** [comment]
Likely related: https://github.com/npm/cli/issues/3159

**Comment by famzah** [comment]
I am astonished that this hasn't been fixed for months! NPM just removed all my global packages.

A workaround is to use `install` instead of `update`.

**Comment by vrodnikov** [comment]
Just faced the same issue with `npm update -g npx`. All symlinks inside `bin/` removed and `lib/node_modules` is empty.

**Environment:**

- OS: Xubuntu 20.04
- Node: 16.13.1
- npm: 8.1.2 (same issue for 8.3.0)

**Comment by mnapoli** [comment]
I reported and closed #4240 as duplicate of this issue. Note that it seems to be the `@` that causes issues, not specifically `@latest`.

I can reproduce it with any version number (all dependencies get wiped out anyway). I also reproduce it with NPM v8.

**Comment by oussamanm** [comment]
is there any way to re-install them , because i dont remember every packages who was installed
